classes/profile.php

```diff
@@ -196,13 +185,121 @@ public function save_record(): int {
 
         // Updates the request_metadata and per reason cache with more recent values.
         if ($this->get('reason') & self::REASON_SLOW) {
-            manager::get_min_duration_for_group_and_reason($this->get('groupby'), self::REASON_SLOW, false);
-            manager::get_min_duration_for_reason(self::REASON_SLOW, false);
+            self::get_min_duration_for_group_and_reason($this->get('groupby'), self::REASON_SLOW, false);
+            self::get_min_duration_for_reason(self::REASON_SLOW, false);
         }
 
         return (int) $this->raw_get('id');
     }
 
+    /**
+     * Returns the minimum duration for profiles matching this reason and page/request.
+     *
+     * Cost: 1 cache read (ideally)
+     * Otherwise: 1 cache read, 1 DB read and 1 cache write.
+     *
+     * @param  string $request holds the request url
+     * @param  int $reason the profile type or REASON_*
+     * @param  bool $usecache whether or not to even bother with caching. This allows for a forceful cache update.
+     *
+     * @return float duration (as seconds) of the fastest profile for a given reason and request/page.
+     */
+    public static function get_min_duration_for_group_and_reason(string $group, int $reason, bool $usecache = true): float {
```

Comment:
Follow on from the comments above around static access. Feels weird to have these be static methods when it could be object methods. I suspect that you will always have a profile object anyway when needing to access these methods, but if not, perhaps it could be refactored to a helper static class?